### PR TITLE
More medicine

### PR DESCRIPTION
**Pull request by ike709**
I've now gone through the entire file. Most `TODO`s couldn't be done because of either my stupidity or something not being implemented. The primary cause of the latter is the lack of `AdjustStun()`, `AdjustKnockdown()`, and `AdjustUnconscious()`.

Most chemicals are fully implemented, but a significant minority are missing `mob_life()`, `reaction_mob()` or `overdose_process()`

Also I'd like to point out that I've tested literally none of the medicine added in this PR or the previous PR. At least it doesn't throw an error when I run the server.

### CODE REVIEW

File: code/modules/reagents/reagents/medicine.js [hunk 1]
```diff
@@ -72,21 +74,21 @@ Object.assign(Inacusiate.prototype, {
 class Cryoxadone extends Medicine { // /datum/reagent/medicine/cryoxadone
 	mob_life(dt) {
 		if(this.holder.c.CarbonMob.bodytemperature >= 0 && this.holder.c.CarbonMob.bodytemperature <= 99) { // At extreme temperatures (upgraded cryo) the effect is greatly increased.
-			//TODO M.status_flags &= ~DISFIGURED
+			this.holder.c.LivingMob.status_flags &= ~combat_defines.DISFIGURED;
```
Comment (ike709) [inline code/modules/reagents/reagents/medicine.js]: Does bitwise stuff work the same in JS as it does in DM?

**Comment by ike709** [comment]
I bet I messed up some `dt` stuff somewhere.

**Comment by ike709** [comment]
`add()` and `remove()` should almost certainly be renamed to `add_reagent()` and `remove_reagent()` at some point. It's also be nice if a `has_reagent()` proc existed to simplify the process of checking for a reagent.

File: code/modules/reagents/reagents/medicine.js [hunk 2]
```diff
@@ -235,7 +254,17 @@ Object.assign(Synthflesh.prototype, {
 	color: [1,0.92,0.92]
 });
 
-class Charcoal extends Medicine {} // /datum/reagent/medicine/charcoal
+class Charcoal extends Medicine { // /datum/reagent/medicine/charcoal
+	mob_life(dt) {
+		this.holder.c.LivingMob.adjust_damage("tox", -1 * dt);
+		for(let reagent of this.holder.c.ReagentHolder.reagents) {
```
[the remark this answers is not in the document]
Comment (monster860) [reply code/modules/reagents/reagents/medicine.js]: Instead of let reagent do `for(let [key, reagent] of ...)` and replace reagent.name with key

**Comment by ike709** [comment]
That's probably right.

File: code/modules/reagents/reagents/medicine.js (hunk 1 above)
Comment (Qustinnus) [reply code/modules/reagents/reagents/medicine.js]: ofcourse it does why would one language change the meaning of the operators 🤔

Comment (ike709) [reply code/modules/reagents/reagents/medicine.js]: because who the fuck knows with DM

**Comment by ike709** [comment]
inb4 there's more fuckups involving `dt`
